### PR TITLE
common yaml functionality

### PR DESCRIPTION
**Pull request by xpander53**

### CODE REVIEW

File: src/main/resources/hl7/resource/Common.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+meta:
```
Comment (klwhaley) [inline src/main/resources/hl7/resource/Common.yml:1]: Add common copyright (see other ymls) for example

File: src/main/resources/hl7/codesystem/ExtensionUrlMapping.yml [hunk 2]
```diff
@@ -19,4 +19,28 @@
...
+- id: "source-event-timestamp"
+  url: "http://ibm.com/fhir/cdm/StructureDefinition/source-event-timestamp"
+# source-record-id
+- id: "source-record-id"
+  url: "http://ibm.com/fhir/cdm/StructureDefinition/source-record-id"
+# source-event-trigger
+- id: "source-event-trigger"
+  url: "http://ibm.com/fhir/cdm/StructureDefinition/source-event-trigger" 
+# source-event-trigger-system
+- id: "source-event-trigger-system"
```
Comment (klwhaley) [inline src/main/resources/hl7/codesystem/ExtensionUrlMapping.yml:39]: This url should be in CodingSystemMapping.yml, as its not an extension definition, but a system url

File: src/main/resources/hl7/codesystem/ExtensionUrlMapping.yml [hunk 3]
```diff
@@ -19,4 +19,28 @@
...
+  url: "http://terminology.hl7.org/CodeSystem/v2-0003" 
+# source-record-type
+- id: "source-record-type"
+  url: "http://ibm.com/fhir/cdm/StructureDefinition/source-record-type" 
+# source-message-type-system
+- id: "source-message-type-system"
```
Comment (klwhaley) [inline src/main/resources/hl7/codesystem/ExtensionUrlMapping.yml:45]: This url should be in CodingSystemMapping.yml, as its not an extension definition, but a system url

File: src/main/resources/hl7/resource/Common.yml (hunk 1 above)
Comment (xpander53) [reply src/main/resources/hl7/resource/Common.yml:1]: Done

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java [hunk 4]
```diff
@@ -330,4 +335,93 @@ private static Practitioner getResourcePractitioner(Resource resource) {
...
+    List<Resource> patientResource =
+        e.stream().filter(v -> ResourceType.Patient == v.getResource().getResourceType())
+            .map(BundleEntryComponent::getResource).collect(Collectors.toList());
+    assertThat(patientResource).hasSize(1);
+    Patient patient = getResourcePatient(patientResource.get(0));
+
+    // Get meta extension from Patient Resource
+    Meta patientMeta = patient.getMeta();
+    List<Extension> patientMetaExtensions = patientMeta.getExtension();
+
```
Comment (LisaWellman) [inline src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java:370]: Add check that we got 6 extensions (need to verify none are missing)

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java [hunk 5]
```diff
@@ -330,4 +335,93 @@ private static Practitioner getResourcePractitioner(Resource resource) {
...
+    // Get the Patient Resource
+    List<Resource> patientResource =
+        e.stream().filter(v -> ResourceType.Patient == v.getResource().getResourceType())
+            .map(BundleEntryComponent::getResource).collect(Collectors.toList());
+    assertThat(patientResource).hasSize(1);
+    Patient patient = getResourcePatient(patientResource.get(0));
+
+    // Get meta extension from Patient Resource
+    Meta patientMeta = patient.getMeta();
+    List<Extension> patientMetaExtensions = patientMeta.getExtension();
```
Comment (LisaWellman) [inline src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java:369]: As we discussed, will want to add checks for other resources besides Patient, so will wnt to move this extension check into a common method.

Will also want to check other message types besides ORU

File: src/main/java/io/github/linuxforhealth/hl7/resource/deserializer/HL7DataBasedResourceDeserializer.java [hunk 6]
```diff
@@ -15,103 +15,125 @@
...
 
   private static final String RESOURCE_TYPE_FIELD_NAME = "resourceType";
   private static final String SPEC = "specs";
+  private static Map<String, Expression> commonExpressions;
 
+  private static final ObjectMapper MAPPER = ObjectMapperUtil.getYAMLInstance();
+  private static final Logger LOGGER = LoggerFactory.getLogger(HL7DataBasedResourceDeserializer.class);
 
+  // Reads the resource/Common.yml and generates expressions from it.
+  private static Map<String, Expression> getCommonExpressions() throws JsonProcessingException {
```
Comment (LisaWellman) [inline src/main/java/io/github/linuxforhealth/hl7/resource/deserializer/HL7DataBasedResourceDeserializer.java:41]: This is not thread-safe.  I think you will need to either:
-- make commonExpressions final, and put this code in a static initializer
-- serialize this method on commonExpressions

File: src/main/resources/hl7/extension/ExtensionMeta.yml [hunk 7]
```diff
@@ -0,0 +1,76 @@
+# lastUpdated:
```
Comment (LisaWellman) [inline src/main/resources/hl7/extension/ExtensionMeta.yml:1]: Remove this

**Review by LisaWellman — COMMENTED** [review]
Looks good, mostly need some more tests

File: src/main/resources/hl7/codesystem/ExtensionUrlMapping.yml (hunk 2 above)
Comment (xpander53) [reply src/main/resources/hl7/codesystem/ExtensionUrlMapping.yml:39]: Done

File: src/main/resources/hl7/codesystem/ExtensionUrlMapping.yml (hunk 3 above)
Comment (xpander53) [reply src/main/resources/hl7/codesystem/ExtensionUrlMapping.yml:45]: Done

File: src/main/resources/hl7/extension/ExtensionMeta.yml (hunk 7 above)
Comment (xpander53) [reply src/main/resources/hl7/extension/ExtensionMeta.yml:1]: Done

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java (hunk 4 above)
Comment (xpander53) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java:370]: Done

File: src/main/java/io/github/linuxforhealth/hl7/resource/deserializer/HL7DataBasedResourceDeserializer.java (hunk 6 above)
Comment (xpander53) [reply src/main/java/io/github/linuxforhealth/hl7/resource/deserializer/HL7DataBasedResourceDeserializer.java:41]: Done

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java (hunk 5 above)
Comment (xpander53) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java:369]: Done

File: src/main/resources/hl7/extension/ExtensionMeta.yml [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+#    valueOf: $timeStamp
+#    constants: 
+#      timeStamp: "2021-03-23T22:09:56.142+00:00"
+
+extension_1:
+   generateList: true
```
[the remark this answers is not in the document]
Comment (xpander53) [reply src/main/resources/hl7/extension/ExtensionMeta.yml:7]: We need this as discussed.

**Comment by LisaWellman** [comment]
Moved to https://github.com/LinuxForHealth/hl7v2-fhir-converter/pull/160

**Pull request closed without merging**